endpoints/integration_test.go

```diff
@@ -196,3 +198,180 @@ func TestReadinessCheck(t *testing.T) {
 		t.Errorf("/status endpoint should always return a 204. Got %d", requestRecorder.Code)
 	}
 }
+
+func TestMultiPutRequestGotStored(t *testing.T) {
+	// Test case: request with more than one element in the "puts" array
+	type aTest struct {
+		description         string
+		elemToPut           string
+		expectedStoredValue string
+	}
+	testCases := []aTest{
+		{
+			description:         "Post in JSON format that contains a bool",
+			elemToPut:           "{\"type\":\"json\",\"value\":true}",
+			expectedStoredValue: "true",
+		},
+		{
+			description:         "Post in XML format containing plain text",
+			elemToPut:           "{\"type\":\"xml\",\"value\":\"plain text\"}",
+			expectedStoredValue: "plain text",
+		},
+		{
+			description:         "Post in XML format containing escaped double quotes",
+			elemToPut:           "{\"type\":\"xml\",\"value\":\"2\"}",
+			expectedStoredValue: "2",
+		},
+	}
+	reqBody := fmt.Sprintf("{\"puts\":[%s, %s, %s]}", testCases[0].elemToPut, testCases[1].elemToPut, testCases[2].elemToPut)
+
+	request, err := http.NewRequest("POST", "/cache", strings.NewReader(reqBody))
+	if err != nil {
+		t.Errorf("Failed to create a POST request: %v", err)
+	}
+
+	//Set up server and run
+	router := httprouter.New()
+	backend := backends.NewMemoryBackend()
+
+	router.POST("/cache", NewPutHandler(backend, 10, true))
+	router.GET("/cache", NewGetHandler(backend, true))
+
+	rr := httptest.NewRecorder()
+
+	router.ServeHTTP(rr, request)
+
+	// validate results
+	var parsed PutResponse
+	err = json.Unmarshal([]byte(rr.Body.String()), &parsed)
+	if err != nil {
+		t.Errorf("Response from POST doesn't conform to the expected format: %s", rr.Body.String())
+	}
+	for i, resp := range parsed.Responses {
+		// Get value for this UUID. It is supposed to have been stored
+		getResult := doMockGet(t, router, resp.UUID)
+
+		// Assertions
+		assert.Equalf(t, http.StatusOK, getResult.Code, "Description: %s \n Multi-element put failed to store:%s \n", testCases[i].description, testCases[i].elemToPut)
+		assert.Equalf(t, testCases[i].expectedStoredValue, getResult.Body.String(), "GET response error. Expected %v. Actual %v", testCases[i].expectedStoredValue, getResult.Body.String())
+	}
+}
+
+func TestEmptyPutRequests(t *testing.T) {
+	// Test case: request with more than one element in the "puts" array
+	type aTest struct {
```

Comment:
Please add a comment indicating that order is extremely important for this test. That's not usually the assumption for table driven tests.